src/machine/cheri.c

```diff
@@ -0,0 +1,621 @@
+/*
+ * Copyright (c) 2024, Capabilities Limited
+ * Copyright (c) 2017 Alex Richardson
+ * Copyright (c) 2014 Robert N. M. Watson
+ *
+ * CHERI support contributed by Capabilities Limited was developed by Hesham Almatary
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/*
+ * This file is copied from CHERI-LLVM's cheri_init_globals.h and modified to
```

Comment:
Why does this need to be modified? Can't you use the compiler-provided header instead. It works for all other usecases? Adding a copy of this code seems like a bad idea.

If the compiler provided header does not work, we should fix that in the compiler and just require a new enough compiler for CHERI.

Reply:
@arichardson It's because seL4 builds standalone and doesn't use or rely on any compiler headers (or standard libraries)

Reply:
It uses `-nostdinc -ffreestanding -fno-common -nostdlib` when building the kernel and doesn't link to the compiler library. But as long as seL4 CHERI support is unverified, tweaking that a bit is preferred to duplicating compiler code.

You did add seL4 specific code to that file though, so it's not a pure copy either.

Reply:
What FreeBSD does to still allow access to the compiler-provided headers is add Clang's resource dir as a explicit include directory, which is quite easy in CMake compared to the complicated makefile logic FreeBSD uses (https://github.com/freebsd/freebsd-src/blob/4b65481ac68a99122991fd73583d071c5e27c65a/share/mk/bsd.sys.mk#L285).

Something like 
```cmake
if (CHERI)
  execute_process(COMMAND ${CMAKE_C_COMPILER} --print-resource-directory OUTPUT_VARIABLE _clang_resource_dir)
  add_compile_options(-idirafter ${_clang_resource_dir}/include)
endif()
```